src/notifications/config.js

```diff
@@ -1,42 +1,43 @@
-const path = require('path');
-const fs = require('fs');
+import path from 'path';
+import fs from 'fs';
 
-/**
- * Construct a configuration object.
- *
- * A configuration object may be constructed with either
- * a path to a `notification.json` file (which defaults to
- * `$PWD/notification.json` if not present, or with a configuration
- * object.
- *
- * @param {String|Object} config Configuration path or details.
- *
- * @constructor
- */
-function Configuration(config) {
-  if (!config) {
-    // I placed the disable line here, however we must not reassign
-    // parameters, this is not allowed, we must fix it later.
-    config = path.join(process.cwd(), 'notification.json'); //eslint-disable-line
+class Configuration {
+  /**
+   * Construct a configuration object.
+   *
+   * A configuration object may be constructed with either
+   * a path to a `notification.json` file (which defaults to
+   * `$PWD/notification.json` if not present, or with a configuration
+   * object.
+   *
+   * @param {String|Object} config Configuration path or details.
+   *
+   * @constructor
+   */
+  constructor(config) {
+    if (!config) {
+      // I placed the disable line here, however we must not reassign
+      // parameters, this is not allowed, we must fix it later.
+      config = path.join(process.cwd(), 'notification.json'); //eslint-disable-line
+    }
+
+    if (typeof config === 'string') {
+      this.loadConfiguration(config);
+    } else {
+      this.configure(config);
+    }
   }
 
-  if (typeof config === 'string') {
-    this.loadConfiguration(config);
-  } else {
+  loadConfiguration(configPath) {
+    const json = fs.readFileSync(configPath);
```

Comment:
Just a suggestion, we could also specify the `encoding` option to true to return a string right away, see [readFileSync](https://nodejs.org/api/fs.html#fs_fs_readfilesync_file_options)

Also I think `json` sounds too generic, maybe `jsonConfig` is better? What do you think?